### PR TITLE
Mosaic create integration test

### LINKED ISSUE #134: Write a simple integration test for creation of an auxiliary chain
1. Run a origin chain
2. Deploy ERC20 token
3 Deploy libraries command
4. create chain command
5. Deploy staker-pool
6. Deploy redeem-pool
7. Call Verify chain command

### PR DESCRIPTION
**Pull request by sarvesh-ost**
fixes #134 

⚠️  Merge after #125 and #131 

This PR adds a integration test to test and demonstrate process of creating a new auxiliary chain.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,11 @@
     "lint": "eslint src -c .eslintrc.json --ext ts",
     "package": "./package.sh",
     "prepack": "npm run package",
-    "test": "npm run test:unit && npm run test:smoke && npm run test:package",
+    "test": "npm run test:unit && npm run test:smoke && npm run test:package && npm run test:integration",
     "test:unit": "mocha -r ts-node/register tests/**/*.test.ts tests/*.test.ts",
     "test:smoke": "./tests/smoke.sh",
     "test:package": "./tests/package.sh",
+    "test:integration": "mocha -r   ts-node/register test_integration/**/*.test.ts test_integration/*.test.ts --timeout 3000000",
```
Comment (abhayks1) [inline package.json]:
```suggestion
    "test:integration": "mocha -r ts-node/register test_integration/**/*.test.ts test_integration/*.test.ts --timeout 3000000",
```

File: src/NewChain/Initialization.ts [hunk 2]
```diff
@@ -162,6 +173,9 @@ export default class Initialization {
       originStateRoot,
     );
 
+    Logger.info('Generating Proof for Stake & mint');
```
Comment (abhayks1) [inline src/NewChain/Initialization.ts]:
```suggestion
    Logger.info('Generated Proof for Stake & mint');
```

File: test_integration/mosaic-create/Utils.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { ChildProcess, spawn } from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import Shell from '../../src/Shell';
+import Directory from '../../src/Directory';
+
+const waitPort = require('wait-port');
+
+export default class Utils {
+  public static startOriginChain(
```
Comment (abhayks1) [inline test_integration/mosaic-create/Utils.ts]: Let's add documentation for the method. Same comment for other methods also.

File: test_integration/mosaic-create/mosaic-create.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    // Second account password.
+    const password = fs.readFileSync(passwordFile).toString().trim().split('\n')[1];
+
+    auxiliaryWeb3.eth.personal.unlockAccount(beneficiary, password);
+    const command = `./mosaic setup-redeem-pool ${originChainId} ${auxChainId} ${auxiliaryEndpoint} ${beneficiary} ${beneficiary} ${beneficiary}`;
+    Shell.executeInShell(command, { stdio: 'inherit' });
+  });
+
+  it(`Verify auxiliary chain ${auxChainId}`, () => {
+    auxiliaryEndpoint = 'http://localhost:40500';
```
Comment (abhayks1) [inline test_integration/mosaic-create/mosaic-create.test.ts]: auxiliaryEndpoint variable can be taken in global scope. It's being used in multiple it.

File: test_integration/mosaic-create/Utils.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    Shell.executeInShell('docker rm integration_test_origin');
+  }
+
+  public static cleanDirectories(originChain: string) {
+    const originDirectory = path.join(
+      Directory.getDefaultMosaicDataDir,
+      originChain,
+    );
+
+    console.log(originDirectory);
```
Comment (abhayks1) [inline test_integration/mosaic-create/Utils.ts]: We should remove console.log ⏰

File: initialize/500.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (puneet-khushwani-eth) [inline initialize/500.json]: Not sure why are we committing this file.

Comment (sarvesh-ost) [reply initialize/500.json]: In order to create a mosaic chain. It expects a config in a particular folder which is (initialise) with name of ${auxiliaryChain}.json name. That's why we need this file.

Comment (puneet-khushwani-eth) [reply initialize/500.json]: Yes please, but question was more why do we need a file for auxiliaryChain:500

**Pull request merged**